artifact/image/layerscanning/image/layer_test.go

```diff
@@ -74,7 +74,7 @@ func TestConvertV1Layer(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			gotLayer, gotError := convertV1Layer(tc.v1Layer, tc.command, tc.isEmpty)
 
-			if tc.wantError != nil && gotError == tc.wantError {
+			if tc.wantError != nil && errors.Is(gotError, tc.wantError) {
```

Comment:
this is where the sole test failure is, and I'm not quite sure what to do - it's the "v1 layer missing diffID" subtest which is expecting the error to involve `ErrDiffIDMissingFromLayer`, so really this is doing `errors.Is(gotError, ErrDiffIDMissingFromLayer)`.

This is how we build that error:

```
return nil, fmt.Errorf("%w: %w", ErrDiffIDMissingFromLayer, err)
```

I think that the failure is legit because we're wanting people to be able to do `errors.Is`, but I'm not sure what the right fix is - attempting to use `errors.Join` doesn't seem to work

Reply:
fwiw I've also confirmed that everything aside from this one test still works fine after #578

Reply:
its because the test is bugged! 🤦 

It was mistakenly asserting that the error _is_ expected error, when it should be asserting its _not_ that error... and ironically it was passing _because it didn't account for wrapped errors_ - so switching to `errors.Is` "fixes" the bug, and thus causes the test to fail...

Reply:
ironically that then reveals that our code is bugged! because we're expecting a `ErrUncompressedReaderMissingFromLayer` but that is never used 😅

Reply:
I've opened #589 to fix the underlying test